src/excmds.ts

```diff
@@ -3281,30 +3281,61 @@ export async function yankimage(url: string): Promise<void> {
 
 /** Change active tab.
 
-    @param index
-        Starts at 1. 0 refers to last tab of the current window, -1 to penultimate tab, etc.
+    @param id
+        A string following the following format: "[0-9]+.[0-9]+", the first number being the index of the window that should be selected and the second one being the index of the tab within that window.
+
+        A bare number means the current window is used.
 
         "#" means the tab that was last accessed in this window
 
-    This is different from [[taball]] because `index` is the position of the tab in the current window.
+        A non integer string means to search the URL and title for matches
  */
 //#background
-export async function tab(index: number | "#") {
-    return tabIndexSetActive(index)
+export async function tab(id: string) {
+    if (Number.isInteger(Number(id)))
+        return tabIndexSetActive(Number(id));
+    if (id === "#")
+        return tabIndexSetActive(id);
+
+    if (id !== null && id !== undefined && !/\d+\.\d+/.exec(id)) {
+        let results = [];
+        try {
+            results = await browser.tabs.query({url: id})
+        } catch (e) { }
+        if (!results.length)
+            results = await browser.tabs.query({title: id.replace("*", "\\*")})
+        if (!results.length) {
+            const urlResult = (await browser.tabs.query({})).find((tab => tab.url.includes(id)))
+            if (urlResult)
+                results = [urlResult]
+        }
+        if (!results.length)
+            results = await browser.tabs.query({title: "*" + id + "*"})
+        if (results.length) {
+            await browser.windows.update(results[0].windowId, { focused: true })
```

Comment:
I'm wondering if the right thing to do in case of multiple matches is to run `fillcmdline_notrail("tabs", query)` to allow the user to pick?

Occasionally in situations like this we give commands an argument to say if they are interactive or not so scripts can still do something useful unattended